src/nrniv/nrndae.cpp

```diff
@@ -203,7 +203,7 @@ void NrnDAE::dkmap(std::vector<neuron::container::data_handle<double>>& pv,
                                                                   y_.data() + i};
         pvdot[bmap_[i] - 1] =
             neuron::container::data_handle<double>{neuron::container::do_not_search,
-                                                   _nt->_actual_rhs + bmap_[i]};
+                                                   _nt->node_rhs_storage() + bmap_[i]};
```

Comment:
Looking at this I'm still not sure how to do that since this constructor needs a `vector` which in that case should be a vector containing all the `rhs` elements?
I'm wondering whether it would make any sense apart from `node_rhs_storage()` to have a function `node_rhs_storage_handle(id)` which returns a `data_handle` already to the specified pointer

Reply:
Just to mention...  I tried the "LinearCircuit:DynamicClamp" neurondemo with address sanitizer for olupton/data-structures-including-mechanisms and it is working well for both "Use variable dt" toggled on and off.

Reply:
extracellular always uses  the sparse13 solver. So does NrnDAE (and LinearMechanism). And so does IDA.

Reply:
I see. I think then this needs to be changed to `_sp13_rhs`. I was wondering whether we could add a test that covers these cases and solvers.
I see from the [code coverage report](https://app.codecov.io/gh/neuronsimulator/nrn/blob/master/src/nrniv/nrndae.cpp) that most of this file is not covered by any test

Reply:
I'll see if a bit of augmentation to the neurondemo test in test_neurondemo.py will increase coverage if the variable step method is used.

Reply:
Hmm. For some reason with 'magkanar/rhs-datastructures-try3' my ubuntu desktop is giving
```
hines@hines-t7500:~/neuron/datahandle/buildcover$ ctest -R test_neurondemo_py -VV
...
62: stderr: /home/hines/neuron/datahandle/buildcover/bin/neurondemo: line 30: 88071 Segmentation fault      (core dumped) ${NRNBIN}nrniv -dll "${NRNDEMO}release/${MODSUBDIR}/.libs/libnrnmech.so" "${NRNDEMO}demo.hoc" "$@" -
62: Traceback (most recent call last):
62:   File "/home/hines/neuron/datahandle/buildcover/test/hoctests/test_neurondemo_py/tests/test_neurondemo.py", line 101, in <module>
62:     data = neurondemo(prgraphs, input % i)
62:   File "/home/hines/neuron/datahandle/buildcover/test/hoctests/test_neurondemo_py/tests/test_neurondemo.py", line 58, in neurondemo
62:     raise Exception("Unexpected stderr")
62: Exception: Unexpected stderr
1/1 Test #62: hoctests::test_neurondemo_py .....***Failed    1.48 sec
```
and olupton/data-structures-including-mechanisms gives
```
hines@hines-t7500:~/neuron/datahandle/buildcover$ ctest -R test_neurondemo_py -VV
...
62: strdiff -40.8301 -40.8302
62: strdiff -41.0783 -41.0785
62: strdiff -41.2185 -41.2189
...
2: demo1 difference
62: std =  [['Graph[0]', [{'x': ['0', '0.025', ...
...
62: Traceback (most recent call last):
62:   File "/home/hines/neuron/datahandle/buildcover/test/hoctests/test_neurondemo_py/tests/test_neurondemo.py", line 129, in <module>
62:     chk(key, rich_data)
62:   File "/home/hines/neuron/datahandle/buildcover/lib/python/neuron/tests/utils/checkresult.py", line 75, in __call__
62:     assert match
62: AssertionError
1/1 Test #62: hoctests::test_neurondemo_py .....***Failed    1.48 sec

0% tests passed, 1 tests failed out of 1
```
Since I'm only trying to get increased coverage for
```
-DNRN_ENABLE_COVERAGE=ON -DNRN_COVERAGE_FILES='src/nrniv/nrndae.cpp;src/nrniv/linmod.cpp;src/nrniv/linmod1.cpp'
```
I'll start over with the master.

Reply:
The master is failing in a way similar to olupton/data-structures-including-mechanisms ... So I guess I'm temporarily diverted from my goal.

Reply:
Thanks a lot for looking into this! If you share somewhere the changes you did to `test_neurondemo_py` I can fix the segmentation fault of my branch

Reply:
I have not made any changes. On my machine test_neurondemo_py fails for master, this pr, and the data_structures-including-mechanisms branch.  test_neurondemo.py has an extra change from Olli in the data_structures branch having to do with `print better diffs when strings don't match` I have not yet been able to find the cause of my problem. If you are having no problem with whatever test_neurondemo.py comes with those branches, then the problem is likely specifically some issue with my machine environment  and probably not very interesting.

Reply:
By the way. Leaving the test failure aside. Using neurondemo starting from 0 coverage and running each demo in turn, gives a base coverage of linmod.cpp 92.3%, linmod1.cpp 69.3%, and nrndae.cpp 77.1% .
Then running neurondemo again and selecting demo(6) and setting cvode_active(1) and running gives new coverages of
linmod.cpp 92.3%, linmod1.cpp 69.3%, and nrndae.cpp 91%

Reply:
> @alexsavulescu and @olupton have a solution for that I think but if I understood correctly it's still work in progress

Planning on having a look tomorrow

Reply:
https://github.com/neuronsimulator/nrn/pull/2174